### DISCUSSION TITLE
Caddy problem

### DISCUSSION THREAD

**Discussion by maxpoz1** [indomus/forum#2272] [HUB: Home Assistant]
Ciao a tutti sono nuovo della vostra comunity e del mondo di Home Assistant o meglio 3 anni fa ho provato ad avere un primo approccio ma poi ho abbandonato perchè per me era troppo complicato quindi arieccomi con i problemi :)

Sto cercando di fare un reverse proxy attraverso una guida presa quì, utilizzo un rpi3 con Home Assistant Os

Il mio problema è che non riesco ad accedere dall'esterno:
Se digito indirizzo http**s**://mioindirizzo.duckdns.org mi esce il messaggio di benvenuto,
se digito https://mioindirizzo.duckdns.org:9000 mi da ERR_SSL_PROTOCOL_ERROR
Se digito indirizzo http://mioindirizzo.duckdns.org mi esce il messaggio di benvenuto

Quindi qualche cosa di giusto l'ho fatto ma non saprei cosa correggere

**Comment by snakuzzo** [comment]
Se non ci dici le configurazioni, è difficile aiutarti.
Comunque, lato addon caddy2 devi semplicemente usare questa configurazione (inserendo ovviamente il tuo dominio e la porta che vuoi usare) e senza usare file sotto `/share/` :

![image](https://user-images.githubusercontent.com/7205371/211909868-fc1d547c-b4d0-47a6-acb3-9b2bec54d214.png)

Su Home Assistant invece, dovrai semplicemente adeguare il blocco `http:` dove vengono specificati i `trusted_proxies` (indicazioni che trovi in guida)

**Reply by maxpoz1** [reply]
QUESTE?

![image](https://user-images.githubusercontent.com/46932447/211916667-b149d427-c539-4035-94a5-bd9af3a2c295.png)

![image](https://user-images.githubusercontent.com/46932447/211916790-c9360b94-a563-422a-b6a3-f499dd5e22f6.png)

![image](https://user-images.githubusercontent.com/46932447/211916971-873ff3fc-e0bc-42f0-939a-e244be2de90b.png)

![image](https://user-images.githubusercontent.com/46932447/211917113-bc1d9811-aa4c-424a-8960-a5044a5e84cb.png)

**Reply by snakuzzo** [reply]
Se ti serve solo HA, non serve tutta quella roba. Serve solo quello che ti ho mostrato io.
Inoltre...non capisco che diavolo hai combinato sul configuration.yaml...è tutto sbagliato e formattato male.

**Reply by snakuzzo** [reply]
Sistema quel file e...se usi un reverse proxy...non devi configurare key e cert nel configuration.yaml di HA, perché se ne occupa il reverse proxy

**Comment by maxpoz1** [comment]
CONFIGURATION YAML
http:
  use_x_forwarded_for: true
  trusted_proxies:
  - 127.0.0.1

Configuraz caddy

destination: localhost
domain: https://xxxxxx.duckdns.org:9000
email: your@email.com
port: 8123

Nulla io non riesco
400: Bad Request

**Reply by snakuzzo** [reply]
nel log di Core vedi qual è l'IP che devi mettere nei `trusted_prosies`.
Ci sei quasi...ti manca solo questo.
Accedi in locale a Core e vedi nei log l'IP che ti segnala...lo metti lì, riavvii ed hai finito

**Reply by maxpoz1** [reply]
non lo so dove si guarda, è l'ip del raspberry?
dove lo metto al posto di 127.0.0.1?

Grazie ancora

**Reply by maxpoz1** [reply]
ti faccio un'altra domanda, il mio router Fritz mi assegna un "ip fisso" tipo "md25rdvo3xscwt.myfritz.net" non posso usare questo al posto di duckdns?

**Reply by snakuzzo** [reply]
Accedi ad HA in locale.
Vai sotto settings > system> logs e troverai n warning per quante volte avrai beccato errore http 400. Lì troverai un IP...quello dovrai metterlo sotto `trusted_proxies`.

Quanto al fritzbox, quello non è un "ip fisso". Non è un ip, ma un fqdn...e sì, puoi usarlo

**Reply by snakuzzo** [reply]
Devi usare il MARKDOWN quando condividi log o codice

**Comment by maxpoz1** [comment]


**Reply by maxpoz1** [reply]
ci sono tanti errori ma non trovo ip

**Comment by maxpoz1** [comment]
```
[homeassistant.components.http.forwarded] Received X-Forwarded-For header from an untrusted proxy ::1**
```

**Reply by snakuzzo** [reply]
Mi sono permesso mi modificare il tuo messaggio, visto che non hai usato il markdown e che hai scritto una sbrodolata di roba che non serve a niente. Ho lasciato solo il messaggio di warning che ci interessa.

Come puoi vedere dal log, Caddy si presenta su HA con `::1` (visto che è abilitato ipv6, si presenta con la loopback ipv6).
Fai così...il blocco `http:` configuralo come indicato di seguito.

```yaml
http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 127.0.0.1   # loopback ipv4
    - ::1  # loopback ipv6
    - 172.30.33.0/24  # subnet di docker (gli addon avranno IP di questa subnet su HA OS, per cui così li trusti tutti)
    - 192.168.1.0/24  # se ti occorre...subnet della tua LAN (da adeguare in base alla classe di indirizzi assegnata dal tuo router)
```

P.S. La prossima volta, se non usi il markdown, il topic verrà cancellato

**Reply by maxpoz1** [reply]
eeeh  ci ho provato di solito nei forum c'è il pulsantino ma non ho trovato quale, poi sono anche andato su MARKDOWN :).
Sono vecchio e rinco....
Adesso provo

**Comment by maxpoz1** [comment]
AZZZ adesso funziona.... eh ci voleva tanto? :)

dai ti ringrazio tanto per ora poi arriverò con l'integrazione di Alexa, per il quale avevo già provato ma non funzionava, probabilmente per questo motivo
Grazie
